README.md

```diff
@@ -1,108 +1,204 @@
+Should the API follow either provider "A" or "B"? Should both approaches be
+implemented, but as optional features? If any of these approaches is defined
+to be a mandatory feature to support, provider "C" cannot be compliant.
+
+Any choice brings significant disadvantages:
+
+- If such features are included as features that are mandatory to support,
+  some providers may have difficulties adopting the API.
+- If such features are included as optional features, the ability to migrate
+  from one provider to another suffers significantly. Without this ability,
+  users also may opt to use provider-specific API's, instead.
+- If such features are excluded, the API becomes overall less useful for the
+  users who may opt to use more powerful provider-specific API's, instead.
+
+In sum: Going this route would be technically the best thing to do, yet does
+not seam feasible given tough trade-offs and limited resources.  
+If the opportunity arises to partner with some other organization with a lot
+of staff and resources, this option may be reevaluated, though.
+
+</details>
+
+## The chosen approach (for now)
+
+Goal: **Provide a "semantically" consistent API modelling most cloud resources
+that are in scope for SCS**.
+
+### Kubernetes API
+
+Instead of creating SCS-specific API idioms and building the implementation
+from scratch, the Kubernetes API will be "reused". Essentially, the Kubernetes
+API is just an opinionated REST API which has opinions on how a resource
+is defined, how it looks like, how it is reconciled/handled, how AuthN/AuthZ
+can be implemented. The Kubernetes ecosystem provides much tooling for working
+with such (custom) resource definitions: For creating the definitions
+themselves, building controllers, making them discoverable and deployable.
+
+As such, Kubernetes is a great choice for building any sort of resource
+management API - with some caveats regarding its deployment and the legacy
+of starting off as container orchestration tooling.
+
+### Crossplane tooling
+
+Crossplane even extends the Kubernetes API with "Composite Resource Definitions"
+(XRD) to make Kubernetes the base for platform engineering within organizations.
+
+Secondly, it provides a API machinery to bring any cloud resource into Kubernetes
+using backend-specific "providers" (roughly comparable with Terraform providers).
+As such, Crossplane with its provider ecosystem actually already did most of
+the heavy lifting for providing e.g. OpenStack resources inside of Kubernetes.
+
+On top, the platform engineering concepts in Crossplane make building multi-tenancy
+systems pretty straight-forward.
+
+Alright. Crossplane takes care of exposing OpenStack resources and does some
+fancy stuff regarding multi-tenancy. What about providing actual Kubernetes
+**workload** clusters?
+
+### Cluster stacks
+
+Cluster stacks are basically an abstraction over CAPI/ClusterAPI. As such, Cluster
```

Comment:
Clusterstacks are mostly tooling that is relevant for the CSP not for the SCS-user. The interesting point for the SCS-User is that there (if we do it right) a documented, public repository/list of clusterclasses (packed in cluster-stacks). Those clusterclasses are of interest for the SCS-User because they define variables that shape the cluster (flavor, dns-settings). Also the choice of the clusterstack/clusterclass will have an impact on the behaviour of the cluster.

Reply:
Ah, alright. Thanks for clarifying. Also had a look at the [cluster-stack-operator docs](https://github.com/SovereignCloudStack/cluster-stack-operator/blob/main/docs/architecture/user-flow.md) now, improving my understanding of cluster stacks. Will change the wording here.

Reply:
Wording is changed.

Reply:
@mxmxchere This can be resolved from your POV?